### PR TITLE
elaborate on what is required for using the tempdirectory setting of nextcloud

### LINKED ISSUE #19682: 'tempdirectory' in config.php is ignored.
### Steps to reproduce
1. Fresh installation of NC
2. Change data directory successfully, config.php looks like:
```
 'datadirectory' => '/mnt/my8TBeZ/nextcloud/data',
  'tempdirectory' => '/mnt/my8TBeZ/nextcloud/data/tmp',
```
3. Upload decently sized file, monitor disk usage during upload. 

### Expected behaviour
'tempdirectory' is used and _only_ my external drive's disk space decreases.

### Actual behaviour
Large (~150MB) chunks are continually written to SD card before being transferred to external.

### Server configuration

**Operating system:**
Raspbian
**Web server:**
Apache
**Database:**
MySQL/MariaDB
**PHP version:**
7.3
**Nextcloud version:** (see Nextcloud admin page)
18.0.1.3
**Updated from an older Nextcloud/ownCloud or fresh install:**
Fresh Install
**Where did you install Nextcloud from:**
NC Github

### Things I've done
Added `upload_tmp_dir = /mnt/my8TBeZ/nextcloud/data/tmp` to my .user.ini file. 
Added `upload_tmp_dir = /mnt/my8TBeZ/nextcloud/data/tmp` to my php.ini file. 

Ensured proper ownership on mount location, running both
```
sudo chmod 0770 /mnt/my8TBeZ/nextcloud/
sudo chown -R www-data:www-data /mnt/my8TBeZ/nextcloud/
```

I can read and write files just fine. The process in which they get there is the part I can't change.

[Apparently I wasn't alone in this](https://github.com/nextcloud/server/issues/9631), I am hoping this thread won't be ignored as much as [@berho](https://github.com/berho)'s was

### PR DESCRIPTION
**Pull request by CaptainSifff**
This elaborates a bit on the tempdirectory setting of nextcloud and how to get it working.

Signed-off-by: Florian Goth <fgoth@physik.uni-wuerzburg.de>

* Resolves: #19682

## TODO

- [ ] Proofread

### CODE REVIEW

**Review by szaimen — APPROVED** [review]

File: config/config.sample.php [hunk 1]
```diff
@@ -1840,7 +1840,10 @@
  * restricted, or if external storage which do not support streaming are in
  * use.
  *
- * The Web server user must have write access to this directory.
+ * The Web server user/PHP must have write access to this directory.
+ * Additionally you have to make sure, that you php configuration considers this a valid
```
Comment (solracsf) [inline config/config.sample.php:1844]:
```suggestion
 * Additionally you have to make sure that your PHP configuration considers this a valid
```

Comment (CaptainSifff) [reply config/config.sample.php:1844]: oh yeah, was too late... thanks!

**Review by kesselb — APPROVED** [review]
I had something different in mind, but fine by me.

**Comment by come-nc** [comment]
I do not think this is true.
Setting the temp directory through environment var is not the same as setting it through tempdirectory config value.
The only requirement should be to allow php to write in the directory.

@CaptainSifff Can you explain this change?

**Comment by CaptainSifff** [comment]
I tried setting the tempdirectory and expected this to be usable by nextcloud. As I found out the hard way, there are more prerequisites like the environment variables of PHP or the configuration of PHP. And beyond that there's apparmor and SELinux. And I wanted to point users to this so that they also check this.

**Comment by come-nc** [comment]
> I tried setting the tempdirectory and expected this to be usable by nextcloud. As I found out the hard way, there are more prerequisites like the environment variables of PHP or the configuration of PHP. And beyond that there's apparmor and SELinux. And I wanted to point users to this so that they also check this.

But it is not the same thing, if you set the env var you do not need to set tempdirectory config option from what I understand.

**Comment by CaptainSifff** [comment]
I think I had to do both, as far as I recall.

**Pull request merged**